### PR TITLE
Documentation improvements

### LINKED ISSUE #129: Inconsistent naming conventions and documentation
There are lot of places in our codebase that have different names or docstrings for essentialy the same things. 
One example is "optimi***z***er" and "optimi***s***er" in `DQNAgent` and `PPOAgent`.

This is not something important but later on we can go over the code base and fix those minor inconsistencies.

### PR DESCRIPTION
**Pull request by maciejors**
- Closes #129;
- Fixes some typos, grammar and logic errors;
- Updates examples so that they don't require horizontal scrolling;
- Adds some missing pieces of documentation, improve existing documentation where it wasn't clear enough;
- Restructures documentation's main page;
- Adds a section on creating custom environments.

### CODE REVIEW

File: academia/agents/dqn_agent.py [hunk 1]
```diff
@@ -67,14 +67,24 @@ class DQNAgent(EpsilonGreedyAgent):
...
+        >>> from academia.utils.models import door_key
         >>>
         >>> # Create an environment:
         >>> env = DoorKey(difficulty=0, append_step_count=True)
-        >>> # Create an instance of the DQNAgent class with custom neural network architecture
-        >>> dqn_agent = DQNAgent(nn_architecture=door_key.MLPStepDQN, n_actions=DoorKey.N_ACTIONS, gamma=0.99,
-        >>>                      epsilon=1.0, epsilon_decay=0.99, min_epsilon=0.01, batch_size=64)
-        >>> # Training loop: Update the agent using experiences (state, action, reward, new_state, done)
+        >>> # Create an instance of the DQNAgent class with
+        >>> #custom neural network architecture
```
Comment (krezelj) [inline academia/agents/dqn_agent.py:76]: Space after #

**Review by krezelj — APPROVED** [review]
Other than one major issue which I believe is unacceptable I think it's all great and good job cause it must have taken a lot of time and effort to do this.

File: academia/agents/dqn_agent.py (hunk 1 above)
Comment (maciejors) [reply academia/agents/dqn_agent.py]: fixed!

**Review by Szymon-Gut — APPROVED** [review]
Looks great! GJ!

**Pull request merged**